opencensus/metrics/export/metric_descriptor.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2018, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class _MetricDescriptorTypeMeta(type):
+    """Helper for `x in MetricDescriptorType`."""
+
+    def __contains__(cls, item):
+        return item in {MetricDescriptorType.GAUGE_INT64,
+                        MetricDescriptorType.GAUGE_DOUBLE,
+                        MetricDescriptorType.GAUGE_DISTRIBUTION,
+                        MetricDescriptorType.CUMULATIVE_INT64,
+                        MetricDescriptorType.CUMULATIVE_DOUBLE,
+                        MetricDescriptorType.CUMULATIVE_DISTRIBUTION}
+
+
+class MetricDescriptorType(object):
+    """The kind of metric. It describes how the data is reported.
+
+    MetricDescriptorType is an enum of valid MetricDescriptor type values. See
+    opencensus-proto for details:
+
+        https://github.com/census-instrumentation/opencensus-proto/blob/24333298e36590ea0716598caacc8959fc393c48/src/opencensus/proto/metrics/v1/metrics.proto#L73  # noqa
+
+    A gauge is an instantaneous measurement of a value.
+
+    A cumulative measurement is a value accumulated over a time interval. In a
+    time series, cumulative measurements should have the same start time and
+    increasing end times, until an event resets the cumulative value to zero
+    and sets a new start time for the following points.
+
+    """
+    __metaclass__ = _MetricDescriptorTypeMeta
```

Comment:
I'm happy to change this if you've got a different preferred way of handling enums.

Reply:
Changed to use `six.add_metaclass` for python 3.